desc/objectives/_coils.py

```diff
@@ -1502,3 +1503,206 @@ def compute(self, field_params=None, constants=None):
+
+    compute::
+
+        w * | K | * | N |
+
+    where K is the winding surface current density, w is the
+    regularization parameter (the weight on this objective),
+    and |N| is the magnitude of the surface normal i.e. the
+    surface jacobian | e_theta x e_zeta |
+
+    This is intended to be used with a surface current::
+
+        K = n x ∇ Φ
+
+    i.e. a CurrentPotentialField
+
+    Intended to be used with a QuadraticFlux objective, to form
+    the REGCOIL algorithm described in [1]_ (if used with a
+    ``FourierCurrentPotentialField``).
+
+    Parameters
+    ----------
+    surface_current_field : CurrentPotentialField
+        Surface current which is producing the magnetic field, the parameters
+        of this will be optimized to minimize the objective.
+    target : {float, ndarray}, optional
+        Target value(s) of the objective. Only used if bounds is None.
+        Must be broadcastable to Objective.dim_f.
+        Defaults to zero.
+    bounds : tuple of {float, ndarray}, optional
+        Lower and upper bounds on the objective. Overrides target.
+        Both bounds must be broadcastable to to Objective.dim_f
+    weight : {float, ndarray}, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        Must be broadcastable to to Objective.dim_f
+        When used with QuadraticFlux objective, this acts as the regularization
+        parameter, with 0 corresponding to no regularization. The larger this
+        parameter is, the less complex the surface current will be, but the
+        worse the normal field.
+    normalize : bool, optional
+        Whether to compute the error in physical units or non-dimensionalize.
+        Note: has no effect on this objective.
+    normalize_target : bool
+        Whether target should be normalized before comparing to computed values.
+        if `normalize` is `True` and the target is in physical units, this should also
+        be set to True.
+        Note: has no effect on this objective.
+    loss_function : {None, 'mean', 'min', 'max'}, optional
+        Loss function to apply to the objective values once computed. This loss function
+        is called on the raw compute value, before any shifting, scaling, or
+        normalization.
+    deriv_mode : {"auto", "fwd", "rev"}
+        Specify how to compute jacobian matrix, either forward mode or reverse mode AD.
+        "auto" selects forward or reverse mode based on the size of the input and output
+        of the objective. Has no effect on self.grad or self.hess which always use
+        reverse mode and forward over reverse mode respectively.
+    source_grid : Grid, optional
+        Collocation grid containing the nodes to evaluate current source at on
+        the winding surface. If used in conjunction with the QuadraticFlux objective,
+        with the same ``source_grid``, this replicates the REGCOIL algorithm described
+        in [1]_.
```

Comment:
I think you mean that this should be the same grid as `field_grid` in the `QuadraticFlux` objective. Because `source_grid` for the flux objective is the virtual casing grid for the plasma component of the magnetic field, which isn't even used for vacuum cases. `field_grid` in the flux objective is used to discretize the magnetic field. 

FYI I think this is used incorrectly in the tutorial notebook (you set `source_grid` when you should set `field_grid` for `QuadraticFlux` in the optimization)

Reply:
I think there are two different ways depending on if it should be the "true" regcoiil method or not:
the true method would require both to have `scalar=True` so that the combined objective is only $\chi^2_B + \lambda \chi^2_K$ in which case it does not actually matter if the source or eval grids for these are the same (and in fact they can be different in the regcoil algorithm as well).

the simpler method that works but is not exactly the regcoil method is not equivalent to using the objectives because it only penalizes Phi_mn, not |K|

Reply:
fixed grid discrepancy in the tutorial

Reply:
techincally, when using the objective function method, it is equivalent to the `"simple"` method, as it essentially is minimizing Bn^2 pointwise (on the eval grid) and lambda*|K|^2 pointwise (on the source grid), which becomes solving a least squares problem for the Phi_mn.

This is NOT equivalent to the `"regcoil"` regularization type as there we are minimizing a single scalar function which is the integral (over eval grid) of Bn^2 plus lambda * integral (over source grid) of |K|^2. To implement this way, we would have to change the `QuadraticFlux` and `SurfaceCurrentRegularization` to return just the integral values, and even then would need to do some weird stuff like returning the sqrt of the integral so that the squared and summed cost for the least squares optimizer is $\chi^2_B + \lambda \chi^2_K$